prov-api/routes/explore/explore.py

```diff
@@ -316,3 +321,58 @@ async def effected_agents(
         record_count=node_count,
         graph=json_serialisation
     )
+
+
+@router.post("/generate/report", response_class = FileResponse, operation_id="export_prov_graph")
+async def generate_report(
+    request: GenerateReportRequest,    
+    background_tasks: BackgroundTasks,
+    roles: ProtectedRole = Depends(read_user_protected_role_dependency),
+    config: Config = Depends(get_settings),  
+) -> FileResponse:
+    
+    """Exports the provenance graph upto a certain upstream depth and fixed downstream depth (1) for Study and Model Run based entities.  
+    This generates a Word Document (.docx) and contains the input, outputs and model runs involved within the Study or within the Model Run. 
+
+    Parameters
+    ----------
+    node_id : str
+        The ID of the starting node for which the report will be generated (Study or Model Run). 
+    depth : int
+        The depth for the upstream direction traversal.
+    item_subtype : ItemSubType
+        The type of the node, which can be either MODEL_RUN or STUDY, determining which report generation path is followed.
+    background_tasks : BackgroundTasks
+        FastAPI background task handler which manage tasks such as cleaning up temporary files. 
+
+    Returns
+    -------
+    FileResponse
+        A FileResponse object that contains the generated word-document and is sent to the front-end. 
+
+    Raises
+    ------
+    HTTPException
+        Raised if the input depth is invalid (not within the range of 1 to 3).
+    HTTPException
+        Raised if an unsupported item subtype is provided.
+    HTTPException
+        Raised if there is an error in generating the word document.
+
+    """
+
+    generated_doc_path:str = await generate_report_helper(
+        node_id=request.id,
+        upstream_depth=request.depth, 
+        item_subtype=request.item_subtype,
+        roles=roles,
+        config=config
+    )
+
+    # Add the removal of the generated word doc as a background task.
+    background_tasks.add_task(remove_file, file_path=generated_doc_path)
```

Comment:
just wanting to check that you can guarantee this task will only delete the file after the file has been streamed to the user?

Reply:
Yep, the deletion task will run only after the response has been sent to the user, according to fast api docs, and haven't had issues in testing as well.